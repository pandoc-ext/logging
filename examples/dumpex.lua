local logging = require 'logging'

local function output(value)
    io.stderr:write(value .. '\n')
end

output(logging.dump(nil))
output(logging.dump(1))
output(logging.dump(false))
output(logging.dump('string'))
output(logging.dump('Hereâ€™s a non-ASCII character.'))
output(logging.dump({}))
output(logging.dump({1, 2, 3}))
output(logging.dump({a=1, b=2, c=3}))
output(logging.dump({a=1, b=2, c=3, 4, 5}))
output(logging.dump({a=1, b=2, c=3, 4, 5, 6, 7, 8, 9, 10}))
output(logging.dump({a=1, b=2, c=3, 4, 5, 6, 7, 8, 9, 10, 11}))
output(logging.dump({1, 2, {3, 4, {a=1, b=2, c=3, d=4, e=5, f=6}}}))
