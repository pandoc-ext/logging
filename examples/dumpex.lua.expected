nil
1
false
"string"
"Hereâ€™s a non-ASCII character."
{}
[3] {[1] 1, [2] 2, [3] 3}
{a: 1, b: 2, c: 3}
[2] {[1] 4, [2] 5, a: 1, b: 2, c: 3}
[7] {
  [1] 4
  [2] 5
  [3] 6
  [4] 7
  [5] 8
  [6] 9
  [7] 10
  a: 1
  b: 2
  c: 3
}
[8] {
  [1] 4
  [2] 5
  [3] 6
  [4] 7
  [5] 8
  [6] 9
  [7] 10
  [8] 11
  a: 1
  b: 2
  c: 3
}
[3] {
  [1] 1
  [2] 2
  [3] [3] {
    [1] 3
    [2] 4
    [3] {
      a: 1
      b: 2
      c: 3
      d: 4
      e: 5
      f: 6
    }
  }
}
